frontend/src/metabase/visualizations/echarts/cartesian/model/axis.ts

```diff
@@ -466,8 +487,32 @@ export function getXAxisModel(
       noRange: isHistogram,
     });
 
+  const minDimensionChange = Math.min(
```

Comment:
This may introduce regressions since the behavior is different from what we had previously. I was working on computing time series interval for the waterfall chart and I saw how it is done for numeric axes as well and reused the older function for numeric axes here https://github.com/metabase/metabase/pull/37908/files#diff-a925af260455ef56ff9bc97d2376ab8cc1f92bc51256d5ab0fdb4b39a9088363R490

We need to be able to specify x-interval by ourselves to handle specific cases we intentionally implemented in the past. Can you reuse this numeric function and make ECharts using our custom x-interval?

Reply:
The original functions behave differently based on the query type: GUI or native. We should also check native queries when testing something like this